[functional_collective] remove the logic that forces torch-xla to use legacy funcol

yifuwang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #123777
* __->__ #123776

After https://github.com/pytorch/xla/pull/6887, torch-xla now also uses
the all_reduce from native funcol. So we can remove this logic.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang

Commit:
[functional_collective] remove the logic that forces torch-xla to use legacy funcol

After https://github.com/pytorch/xla/pull/6887, torch-xla now also uses
the all_reduce from native funcol. So we can remove this logic.

[ghstack-poisoned]

Commit:
Update on "[functional_collective] remove the logic that forces torch-xla to use legacy funcol"


After https://github.com/pytorch/xla/pull/6887, torch-xla now also uses
the all_reduce from native funcol. So we can remove this logic.

[ghstack-poisoned]

wanchaol reviewed (approved):
Nice! I wonder if we should update the xla pin to include the xla side of changes? cc @alanwaketan

alanwaketan commented:
> Nice! I wonder if we should update the xla pin to include the xla side of changes? cc @alanwaketan

Let's do it. Just grep xla.txt and use the hash of the head of torch-xla.

Commit:
Update on "[functional_collective] remove the logic that forces torch-xla to use legacy funcol"


After https://github.com/pytorch/xla/pull/6887, torch-xla now also uses
the all_reduce from native funcol. So we can remove this logic.

[ghstack-poisoned]
